waspc/data/Generator/templates/server/src/auth/hooks.ts

```diff
@@ -0,0 +1,39 @@
+{{={= =}=}}
+import { prisma } from 'wasp/server'
+import { type ProviderId } from 'wasp/auth/utils'
+import { type AuthHookFn } from 'wasp/auth/hooks'
+
+{=# onUserSignUpHook.isDefined =}
+{=& onUserSignUpHook.importStatement =}
+export const onUserSignUpHook = addExtraParamsToTheHookCall({= onUserSignUpHook.importIdentifier =})
+{=/ onUserSignUpHook.isDefined =}
+{=^ onUserSignUpHook.isDefined =}
+export const onUserSignUpHook = undefined
+{=/ onUserSignUpHook.isDefined =}
+
+{=# onUserLoginHook.isDefined =}
+{=& onUserLoginHook.importStatement =}
+export const onUserLoginHook = addExtraParamsToTheHookCall({= onUserLoginHook.importIdentifier =})
+{=/ onUserLoginHook.isDefined =}
+{=^ onUserLoginHook.isDefined =}
+export const onUserLoginHook = undefined
+{=/ onUserLoginHook.isDefined =}
+
+{=# onUserLogoutHook.isDefined =}
+{=& onUserLogoutHook.importStatement =}
+export const onUserLogoutHook = addExtraParamsToTheHookCall({= onUserLogoutHook.importIdentifier =})
+{=/ onUserLogoutHook.isDefined =}
+{=^ onUserLogoutHook.isDefined =}
+export const onUserLogout = undefined
+{=/ onUserLogout.isDefined =}
+
+type InternalAuthHookFn = ({ providerId }: { providerId: ProviderId }) => Promise<void> | void
+
+function addExtraParamsToTheHookCall(hookFn: AuthHookFn): InternalAuthHookFn {
```

Comment:
So you say this adds extra params, but it seems to me like it removes params -> it takes user's hook function, which takes providerId and prisma, and partially applies prisma to it, returning a function that takes only providerId, which is now internal auth hook fn.

Aha, maybe when you said "addExtraParams" you meant "applyExtraParams"? Btw we could go further and say "applyExtraArgs", since parameters are variables in the function declaration, and arguments are values passed to them.

Maybe `userAuthHookToInternalAuthHook` might also be a good name btw.